Publishing HTML Reports in Pipeline

bitwiseman opened the pull request:
@rtyler - Ready for review

rtyler commented on content/blog/2016/2016-06-30-html-publisher-plugin.adoc:
```diff
@@ -0,0 +1,133 @@
+I've found a ruby project,
+link:https://github.com/reiseburo/hermann[hermann], I'd like to build using Jenkins Pipeline. I'd
+also like to have the code coverage results published with each build job.  I could
+write a plugin to publish this data, but I'm in a bit of hurry and it the build
```
> it the build

I think this is supposed to say:

> it appears that the build already craetes an html report file

rtyler commented on content/blog/2016/2016-06-30-html-publisher-plugin.adoc:
```diff
@@ -0,0 +1,133 @@
+
+=== Let's Make This Quick
+
+I've found a ruby project,
```
Ruby is always capitalized

rtyler commented on content/blog/2016/2016-06-30-html-publisher-plugin.adoc:
```diff
@@ -0,0 +1,133 @@
+----
+
+Simple enough, it builds, runs tests, and archives the package.
+image::/images/post-images/2016-06-30/run-1.png[Job Run Without Report Link, role=center]
```
I think this needs a newline above it for the `image` macro to be rendered properly (GitHub doesn't think there's an image here)

rtyler commented on content/blog/2016/2016-06-30-html-publisher-plugin.adoc:
```diff
@@ -0,0 +1,133 @@
+
+image::/images/post-images/2016-06-30/snippet-generator-1.png[Snippet Generator Menu, role=center]
+
+The it shows me a UI similar to the one used in job configuration.  I fill in
```
> The it shows

should probably be

> Then it shows me

rtyler commented on content/blog/2016/2016-06-30-html-publisher-plugin.adoc:
```diff
@@ -0,0 +1,133 @@
+
+=== HTML Published
+My use case is simple, so I'll just reformat it and copy it in at the end of my
+pipeline.  (I ran into a bug in the snippet generated for this step, but typing
```
What bug? Was it reported upstream already?

rtyler replied:
I'm assuming this is what is referenced in the code comments below, I suggest noting it and linking it in this statement in some positive fashion? :confused:

rtyler commented on content/blog/2016/2016-06-30-html-publisher-plugin.adoc:
```diff
@@ -0,0 +1,133 @@
+I can use that directly or as a template for further customization.
+
+=== HTML Published
+My use case is simple, so I'll just reformat it and copy it in at the end of my
```
AFAICT "use-case" should by hypenated, I could be wrong there

bitwiseman replied:
Google doesn't turn up hyphenated cases, but visually I agree it looks like it needs hyphenation.

rtyler commented on content/blog/2016/2016-06-30-html-publisher-plugin.adoc:
```diff
@@ -0,0 +1,133 @@
+----
+
+When I run this new pipeline I am rewarded with an `RCov Report` link on left side,
+which I can follow to show the HTML report.
```
There are a few strings where it's "html report" and a few where it's "HTML report". I think it should be consistent throughout the post as "HTML report"

bitwiseman replied:
Absolutely.

rtyler commented on content/blog/2016/2016-06-30-html-publisher-plugin.adoc:
```diff
@@ -0,0 +1,133 @@
+=== Links
+
+link:https://wiki.jenkins-ci.org/display/JENKINS/HTML+Publisher+Plugin[HTML Publisher Plugin]
+link:https://jenkins.io/blog/2016/05/31/pipeline-snippetizer/[Jenkins Pipeline Snippet Generator]
```
Mind making this jenkins.io link a relative link?

rtyler commented:
I'm not sure where to put it, but I think it might be worth referencing _how_ the Ruby project creates coverage reports. They're generated via [SimpleCov](https://github.com/colszowka/simplecov) when [the unit tests are run](https://github.com/reiseburo/hermann/blob/master/spec/spec_helper.rb#L8).

oleg-nenashev commented on content/blog/2016/2016-06-30-html-publisher-plugin.adoc:
```diff
@@ -0,0 +1,133 @@
+How do I add the HTML publishing step to the pipeline?  The plugin page doesn't
+say anything about it.
+
+=== Snippet Generator to the Rescue
```
🐛 Snippet generator works incorrectly for this step. It's a known issue in Pipeline

oleg-nenashev replied:
FTR https://issues.jenkins-ci.org/browse/JENKINS-29711

bitwiseman replied:
I linked to the bug.

oleg-nenashev replied:
Screenshot shows wrong snippet from what I see

bitwiseman replied:
Wrong? I think it shows the snippet that was generated. I then note that I modified it.

oleg-nenashev commented:
> I'm not sure where to put it, but I think it might be worth referencing how the Ruby project creates coverage reports. They're generated via SimpleCov when the unit tests are run.

Be accurate with it, because since Content SECURITY fixes several use-cases are broken by default (e.g. publishing of Doxygen reports). Needs a kind of disclaimer then

bitwiseman commented:
Re: Content Security - These reports seem to be working, but if you think it is important I can drill in and add more to the post, including Content Security.

rtyler commented:
> Be accurate with it, because since Content SECURITY fixes several use-cases are broken by default (e.g. publishing of Doxygen reports). Needs a kind of disclaimer then

@oleg-nenashev I have no idea what you're referring to here

bitwiseman commented:
@rtyler - Addressed your comments, added some style cleanup, 
The Content Security oleg refers to is the warning section at the top of the page for https://wiki.jenkins-ci.org/display/JENKINS/HTML+Publisher+Plugin: 

> Starting in versions 1.625.3 and 1.641, Jenkins restricted what kind of content could be displayed when serving static files. This can impact how HTML files archived using this plugin are displayed. See Configuring Content Security Policy for more information.

oleg-nenashev commented:
@rtyler @bitwiseman 

Yeah, my sentences are not very clear sometimes :(

Several references FYI:
* Advisory: https://wiki.jenkins-ci.org/display/SECURITY/Jenkins+Security+Advisory+2015-12-09
* Partial fix by @daniel-beck : https://github.com/jenkinsci/htmlpublisher-plugin/pull/22
